src/GafferImage/ImageWriter.cpp

```diff
@@ -841,6 +874,43 @@ const std::string ImageWriter::currentFileFormat() const
 	}
 }
 
+void ImageWriter::registerDefaultColorSpace( DefaultColorSpaceFunction f )
+{
+	defaultColorSpaceFunc() = f;
+}
+
+ImageWriter::DefaultColorSpaceFunction &ImageWriter::defaultColorSpaceFunc()
+{
+	static DefaultColorSpaceFunction g_colorSpaceFunction;
+	return g_colorSpaceFunction;
+}
+
+const std::string ImageWriter::defaultColorSpace() const
+{
+	const std::string fileFormatName = currentFileFormat();
+	const ValuePlug *optionsPlug = this->getChild<ValuePlug>( fileFormatName );
+
+	if( optionsPlug == NULL)
+	{
+		return "";
```

Comment:
Sure, but for any format that supports multiple data types, we should be providing an option plug, and even for formats which don't support multiple data types, the colourspace function may still have an opinion on what the colourspace should be, based on other knowledge of the format. Not calling the colourspace function at all means it simply has no chance...